### ISSUE TITLE
中间状态的 tensor 的 stop_gradient = True\False 有什么影响？

### ISSUE THREAD

**Issue by Sky-zzt** [PaddlePaddle/Paddle#56183]
### 请提出你的问题 Please ask your question

中间状态的 tensor 的 stop_gradient = True\False 有什么影响？
1、比如我通过 nonzero、index_select 等操作得到的中间tensor，默认stop_gradient = False 。我怎么确确认是不是应该设置为stop_gradient = True
比如：
```
        self.faxian_ins=paddle.nonzero(self.faxian_ins_mask,as_tuple=True)[0]
        self.cast_fx_label=paddle.index_select(self.cast_label, self.faxian_ins)
        self.fx_logit=paddle.index_select(self.faxian_logits, self.faxian_ins)
        self.fx_ins_weight=paddle.index_select(self.ins_weight, self.faxian_ins)
//上面几个就是中间tensor 打印出来默认是stop_gradient = False ，那是不是说就会更新这几个tensor？

//下面是loss 计算逻辑
        def true_func_fx():
            ones_label_1 = paddle.ones(paddle.shape(self.cast_fx_label),dtype="float32",name= "ones_label_1")
            zeros_label_1 = paddle.zeros(paddle.shape(self.cast_fx_label),dtype="float32",name= "zeros_label_1")
            mask_label_one_1 = paddle.equal(ones_label_1,self.cast_fx_label,name= "mask_label_one_1")
            mask_label_zero_1 = paddle.equal(zeros_label_1,self.cast_fx_label,name= "mask_label_zero_1")
            pos_1 = paddle.masked_select(self.fx_logit,mask_label_one_1,name= "pos_1")
            neg_1 = paddle.masked_select(self.fx_logit,mask_label_zero_1,name= "neg_1")
            pos1 = pos_1.reshape([paddle.shape(pos_1)[0],1])
            neg1 = neg_1.reshape([paddle.shape(neg_1)[0],1])
            pos_weight = paddle.masked_select(self.fx_ins_weight,mask_label_one_1,name= "pos_weight")
            neg_weight = paddle.masked_select(self.fx_ins_weight,mask_label_zero_1,name= "neg_weight")
            pos_weight1 = pos_weight.reshape([paddle.shape(pos_weight)[0],1])
            neg_weight1 = neg_weight.reshape([paddle.shape(neg_weight)[0],1])
            rand_pos = paddle.expand(pos_weight1, shape=[pos_weight1.shape[0], neg_weight1.shape[0]])
            rand_neg = paddle.expand(neg_weight1, shape=[neg_weight1.shape[0], pos_weight1.shape[0]]).T
            add_weight = paddle.add(rand_pos,rand_neg)
            pos_out = paddle.expand(pos1, shape=[paddle.shape(pos1)[0], paddle.shape(neg1)[0]],name='expand')
            new_t = paddle.log(F.sigmoid(paddle.clip(pos_out - neg1.T,min=-15.0, max=15.0)))
            new_t = paddle.multiply(new_t, add_weight)
            new_t = paddle.clip(new_t,min=-150.0, max=150.0)
            return -paddle.mean(new_t)  //loss 计算

        self.fx_avg_cost = paddle.static.nn.cond(cond_fx,true_func_fx,false_func_fx) //loss 

```

**Comment by w5688414** [comment]
可以参考一下这个 [stop_gradient 为True](https://www.paddlepaddle.org.cn/documentation/docs/zh/faq/train_cn.html#stop-gradient-true)
